### PR TITLE
add Ed448 support

### LINKED ISSUE #71: Add Ed448 support
We want to support all curves defined in the spec.

### PR DESCRIPTION
**Pull request by conradoplg**
Closes #71 

This is currently pointing `Ed448-Goldilocks` to the branch in our fork corresponding to the PR https://github.com/crate-crypto/Ed448-Goldilocks/pull/23 . It will need to be updated it upstream merges or if we decided to merge in our fork.

This also removes the `Default` bound from `Serialization` because it's not strictly needed and makes it harder to support Ed448, since arrays larger than 256 bytes do not implement it.

### CODE REVIEW

File: frost-ed448/tests/frost.rs
Comment (dconnolly) [inline frost-ed448/tests/frost.rs:21]: 👍

File: frost-ed448/tests/frost.rs
Comment (dconnolly) [inline frost-ed448/tests/frost.rs:30]: 👍

File: frost-core/src/signature.rs
Comment (dconnolly) [inline frost-core/src/signature.rs:37]: Why `one`, and not `zero()`? That's basically what `Serialization::default()` did before

File: frost-ed448/src/lib.rs
Comment (dconnolly) [inline frost-ed448/src/lib.rs:53]: Is this a workaround until the upstream fixes are merged?

File: frost-ed448/src/lib.rs
Comment (dconnolly) [inline frost-ed448/src/lib.rs:92]: Funny they don't have a proper means of getting these bytes out

**Review by dconnolly — APPROVED** [review]
🎉

A comment or two but no blockers 💃

File: frost-ed448/src/lib.rs
Comment (conradoplg) [reply frost-ed448/src/lib.rs:53]: Ed448 is tricky because points are encoded as 57 bytes (56 full bytes for the field element, forcing an extra byte for the sign). But scalars can be encoded as 56 bytes. I'm working on a PR upstream to make `Scalar::to_bytes()` return 57 bytes but it does make it weird because Ed448-Goldilocks also supports X448 and in that scenario there is no reason to encode scalars as 57 bytes and all test vectors have 56 bytes :smiling_face_with_tear: 

Anyway I'll check with the maintainer what seems to be the best solution, feedback is welcome

File: frost-ed448/src/lib.rs
Comment (conradoplg) [reply frost-ed448/src/lib.rs:92]: To be fair `curve25519-dalek`, while it has `to/as_bytes`, also makes the bytes `pub`.

I want to eventually try to make the `Ed448-Goldilocks` API as similar as `curve25519-dalek` as possible

File: frost-core/src/signature.rs
Comment (conradoplg) [reply frost-core/src/signature.rs]: Good point, no idea why I picked `one()`. Changed to  `zero()` in 17c80c1

File: frost-ed448/src/lib.rs
Comment (dconnolly) [reply frost-ed448/src/lib.rs:53]: Weird, not my favorite, but oh well!

File: frost-core/src/signature.rs
Comment (dconnolly) [reply frost-core/src/signature.rs]: Nice 😎

**Review by dconnolly — APPROVED** [review]
looks good! Thanks for answering my questions :D

**Pull request merged**
